### PR TITLE
feat: support admission.k8s.io/v1 alongside v1beta1

### LINKED ISSUE #678: Feature: Support Admission Request and review v1 API
The admission webhook currently supports v1beta1 APIs.

### PR DESCRIPTION
**Pull request by mflendrich**
**What this PR does / why we need it**:

This PR:
- upgrades the `admission.k8s.io` APIs used by KIC to `v1` from `v1beta1`. This PR assumes that a `v1` webhook is backwards compatible with a `v1beta1` apiserver as its caller. In other words, the webhook will accept both `v1beta1` and `v1` requests.
- updates `ValidationWebhookConfiguration` manifests to allow `v1` APIs (k8s >= 1.16) alongside `v1beta1`.
- refactors the test for the admission controller server to [table-driven](https://github.com/golang/go/wiki/TableDrivenTests) for the following benefits:
  - makes it possible to introduce `v1` tests alongside `v1beta1` without exponential code growth
  - streamlined test execution (all tests have the same workflow but different inputs and expected outputs)
  - table-driven tests are idiomatic Go
- run each server test for `v1beta1` as well as `v1`

**Which issue this PR fixes**

fixes #678 

**Special notes for your reviewer**:
Leaves `ValidationWebhookConfiguration` at `v1beta1` for compatibility with k8s < 1.16. Added TODOs for a later cleanup.

### CODE REVIEW

**Comment by hbagdi** [comment]
As discussed out of band, we need v1beta1 and v1 at the same time.

BTW, nice work with table driven test refactoring here.
Make sure to have two separate commits, one for tests refactoring and other adding the feature, helps in git spellunking in future. Thanks!

**Comment by mflendrich** [comment]
> As discussed out of band, we need v1beta1 and v1 at the same time.

This PR supports both versions, according to my understanding of the `admission.k8s.io` documentation about (minimal) changes between these versions - `v1` being essentially a promoted `v1beta1` with some stricter validations. Reworded the PR description to capture it better.

> BTW, nice work with table driven test refactoring here.
> Make sure to have two separate commits, one for tests refactoring and other adding the feature, helps in git spellunking in future. Thanks!

Thanks! These are separate commits already.

**Comment by mflendrich** [comment]
KIC admission webhook manually verified to work with:
- kubernetes `1.15` and `admission.k8s.io/v1beta1`
- kubernetes `1.18` and `admission.k8s.io/v1`

File: deploy/manifests/validation-webhook-configuration.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ---
-apiVersion: admissionregistration.k8s.io/v1beta1
+apiVersion: admissionregistration.k8s.io/v1beta1  # TODO: change to v1 when we no longer support k8s <1.16
```
Comment (hbagdi) [inline deploy/manifests/validation-webhook-configuration.yaml:2]: These are user facing manifests and scripts. Can we instead open a Github issue to track this and remove this TODO?
Having TODO in code is okay but having it here _could_ lead to confusion for end-users.
Not a blocker for merge but good to have.

File: internal/admission/server_test.go [hunk 2]
```diff
@@ -55,26 +56,25 @@ func TestServeHTTPBasic(t *testing.T) {
...
 
-		wantRespCode        int
-		wantSuccessResponse admission.AdmissionResponse
-		wantFailureMessage  string
-	}{
-		{
-			name:               "request with present empty body",
-			wantRespCode:       http.StatusBadRequest,
-			wantFailureMessage: "unexpected end of JSON input\n",
-		},
```
Comment (hbagdi) [inline internal/admission/server_test.go]: Question: Why remove this test case?

File: deploy/manifests/validation-webhook-configuration.yaml (hunk 1 above)
Comment (mflendrich) [reply deploy/manifests/validation-webhook-configuration.yaml]: Done. #763

File: internal/admission/server_test.go (hunk 2 above)
Comment (mflendrich) [reply internal/admission/server_test.go]: Accidentally 🤦. Fixed.

**Comment by mflendrich** [comment]
Please take a look now @hbagdi. Added two new coimmits - to be autosquashed before merging.

**Comment by mflendrich** [comment]
done, looks like it requires another approval now 🙄 

@hbagdi do we want to wait for #764 and then rebase on top of the new `next`?

**Comment by hbagdi** [comment]
> @hbagdi do we want to wait for #764 and then rebase on top of the new next?

Sorry, I saw this comment after I merged 764 else I'd have merged this one first. Can you do a rebase please?

**Comment by mflendrich** [comment]
> Can you do a rebase please?

Done.
From my perspective, the PR is ready for merging.

**Pull request merged**